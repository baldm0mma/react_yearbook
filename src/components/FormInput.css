.form-input {
  display: flex;
  flex-direction: column;
  width: 30%;
  align-items: center;
  margin: 0 auto;
  justify-content: center;
}

button {
  margin: 10px auto;
}