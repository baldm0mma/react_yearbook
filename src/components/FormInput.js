import React, { Component } from 'react';
import './FormInput.css';

class FormInput extends Component {
  constructor() {
    super();
    this.state = {
      id: 0,
      name: '',
      quote: '',
      superlative: '',
      status: ''
    };
  }

  handleStateChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
    console.log(this.state.status);
  };

  handleSubmit = e => {
    e.preventDefault();
    this.props.addNewHuman(this.state);
    console.log(this.state.name);
  };

  render() {
    const { name, quote, superlative } = this.state;
    return (
      <>
        <h2>Add a human</h2>
        <form className='form-input'>
          <input
            type='text'
            name='name'
            value={name}
            placeholder='Name'
            onChange={this.handleStateChange}
          />
          <input
            type='text'
            name='quote'
            value={quote}
            placeholder='Quote'
            onChange={this.handleStateChange}
          />
          <input
            type='text'
            name='superlative'
            value={superlative}
            placeholder='Superlative'
            onChange={this.handleStateChange}
          />
          <select>
            <option
              name='status'
              value='staff'
              onChange={this.handleStateChange}
            >
              Staff
            </option>
            <option
              name='status'
              value='student'
              onChange={this.handleStateChange}
            >
              Student
            </option>
          </select>
          <button onClick={this.handleSubmit}>Submit!</button>
        </form>
      </>
    );
  }
}

export default FormInput;
