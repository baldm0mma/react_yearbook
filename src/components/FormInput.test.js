import React from 'react';
import { shallow, mount } from 'enzyme';
import FormInput from './FormInput';

describe('FormInput', () => {
  it('should render the correct elements with the correct information', () => {
    const wrapper = shallow(<FormInput />);

    expect(wrapper).toMatchSnapshot();
  });
});