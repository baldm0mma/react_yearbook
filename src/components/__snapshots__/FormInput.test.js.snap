// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput should render the correct elements with the correct information 1`] = `
<Fragment>
  <h2>
    Add a human
  </h2>
  <form
    className="form-input ui input"
  >
    <input
      className="input-style"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <input
      className="input-style"
      name="quote"
      onChange={[Function]}
      placeholder="Quote"
      type="text"
      value=""
    />
    <input
      className="input-style"
      name="superlative"
      onChange={[Function]}
      placeholder="Superlative"
      type="text"
      value=""
    />
    <select
      name="status"
      onChange={[Function]}
    >
      <option
        value="staff"
      >
        Staff
      </option>
      <option
        value="students"
      >
        Student
      </option>
    </select>
    <button
      onClick={[Function]}
    >
      Submit!
    </button>
  </form>
</Fragment>
`;
